### PR TITLE
JAV-524 Optimize edge executor

### PR DESCRIPTION
**Pull request by wujimin**
use the same mechanism of normal microservice producer

before optimize, edge's decode and filter run in net thread.

### CODE REVIEW

File: common/common-rest/src/main/java/io/servicecomb/common/rest/RestProducerInvocation.java [hunk 1]
```diff
@@ -65,115 +52,26 @@ public void invoke(Transport transport, HttpServletRequestEx requestEx, HttpServ
...
-      QueueMetricsData reqQueue = MetricsServoRegistry.getOrCreateLocalMetrics()
-          .getOrCreateQueueMetrics(restOperationMeta.getOperationMeta().getMicroserviceQualifiedName());
-      reqQueue.incrementTotalCount();
-      Long timeInQueue = metricsData.getQueueEndTime() - metricsData.getQueueStartTime();
-      reqQueue.setTotalTime(reqQueue.getTotalTime() + timeInQueue);
-      reqQueue.setMinLifeTimeInQueue(timeInQueue);
-      reqQueue.setMaxLifeTimeInQueue(timeInQueue);
-      reqQueue.decrementCountInQueue();
-    }
+  protected OperationLocator locateOperation(ServicePathManager servicePathManager) {
```
Comment (WillemJiang) [inline common/common-rest/src/main/java/io/servicecomb/common/rest/RestProducerInvocation.java]: What will happen if we cannot find the OperationLocator?

Comment (wujimin) [reply common/common-rest/src/main/java/io/servicecomb/common/rest/RestProducerInvocation.java]: if locate failed, throw 404 or 405 exception.

File: transports/transport-rest/transport-rest-servlet/src/main/java/io/servicecomb/transport/rest/servlet/RestServletProducerInvocation.java [hunk 2]
```diff
@@ -17,20 +17,17 @@
...
 import io.servicecomb.foundation.vertx.http.StandardHttpServletRequestEx;
 
 public class RestServletProducerInvocation extends RestProducerInvocation {
   @Override
-  protected RestOperationMeta findRestOperation() {
-    RestOperationMeta restOperationMeta = super.findRestOperation();
+  protected void findRestOperation() {
+    super.findRestOperation();
 
     boolean cacheRequest = collectCacheRequest(restOperationMeta.getOperationMeta());
```
Comment (WillemJiang) [inline transports/transport-rest/transport-rest-servlet/src/main/java/io/servicecomb/transport/rest/servlet/RestServletProducerInvocation.java]: this variable is useless.

File: transports/transport-rest/transport-rest-servlet/src/main/java/io/servicecomb/transport/rest/servlet/RestServletProducerInvocation.java [hunk 3]
```diff
@@ -17,20 +17,17 @@
...
 
 import io.servicecomb.common.rest.RestProducerInvocation;
-import io.servicecomb.common.rest.definition.RestOperationMeta;
 import io.servicecomb.common.rest.filter.HttpServerFilter;
 import io.servicecomb.core.definition.OperationMeta;
 import io.servicecomb.foundation.vertx.http.StandardHttpServletRequestEx;
 
 public class RestServletProducerInvocation extends RestProducerInvocation {
   @Override
-  protected RestOperationMeta findRestOperation() {
```
Comment (WillemJiang) [inline transports/transport-rest/transport-rest-servlet/src/main/java/io/servicecomb/transport/rest/servlet/RestServletProducerInvocation.java]: It looks like the new findRestOperation method is just throw the exception if it cannot find the operation.  I'm not sure if this change has some side effect on the user side code.

File: transports/transport-rest/transport-rest-servlet/src/main/java/io/servicecomb/transport/rest/servlet/RestServletProducerInvocation.java (hunk 2 above)
Comment (wujimin) [reply transports/transport-rest/transport-rest-servlet/src/main/java/io/servicecomb/transport/rest/servlet/RestServletProducerInvocation.java]: which one?
cacheRequst?   it's setted to requestEx,  if copy request body controlled by this avriable.

File: transports/transport-rest/transport-rest-servlet/src/main/java/io/servicecomb/transport/rest/servlet/RestServletProducerInvocation.java (hunk 3 above)
Comment (wujimin) [reply transports/transport-rest/transport-rest-servlet/src/main/java/io/servicecomb/transport/rest/servlet/RestServletProducerInvocation.java]: not change, just move from one class to another class.

File: transports/transport-rest/transport-rest-servlet/src/main/java/io/servicecomb/transport/rest/servlet/RestServletProducerInvocation.java (hunk 2 above)
Comment (WillemJiang) [reply transports/transport-rest/transport-rest-servlet/src/main/java/io/servicecomb/transport/rest/servlet/RestServletProducerInvocation.java]: I missed the second line , it makes sense now.

**Pull request merged**